src/ketu/async/source.clj

```diff
@@ -125,7 +126,10 @@
 
             (loop []
               (when-let [records (poll!)]
-                (run! put! records)
+                (if (seq records)
+                  (run! put! records)
+                  ; Throws exception if not connected
```

Comment:
do we want automatic recovery? or does it belong in the application / wrapper library?